pkg/upstream/replicated.go

```diff
@@ -447,6 +467,13 @@ func downloadReplicatedApp(replicatedUpstream *replicatedapp.ReplicatedUpstream,
 		case tar.TypeReg:
 			content, err := io.ReadAll(tarReader)
 			if err != nil {
+				io.ReadAll(cachingBodyReader)
+				cachedContent := cachingBodyReader.GetCachedBytes()
+				readableText := util.ExtractReadableText(cachedContent)
+				if len(readableText) > 0 {
+					logger.Info("failed to read file from tar",
+						zap.String("readable_text", readableText))
+				}
```

Comment:
can you move this code to a util function? `maybeLogReadableText` or something like that? maybe even make it a function of `cachingBodyReader`?

Reply:
Sure